vconfig/viper.go

```diff
@@ -66,3 +82,31 @@ func setWebsocketDefaults() {
 	viper.SetDefault(EnvWebsocketHistoryMax, 300)
 	viper.SetDefault(EnvWebsocketHistoryTtl, 10)
 }
+func setP2PDefaults() {
+	viper.SetDefault(EnvP2PLogLevel, p2pconfig.DefaultLogLevel)
+	viper.SetDefault(EnvP2PMaxPeers, p2pconfig.DefaultMaxPeers)
+	viper.SetDefault(EnvP2PMaxPeersPerIP, p2pconfig.DefaultMaxPeersPerIP)
+	viper.SetDefault(EnvP2PMinSyncPeerNetworkSpeed, p2pconfig.DefaultMinSyncPeerNetworkSpeed)
+	viper.SetDefault(EnvP2PBanDuration, p2pconfig.DefaultBanDuration)
+	viper.SetDefault(EnvP2PLogDir, p2pconfig.DefaultLogDir)
+	viper.SetDefault(EnvP2PExcessiveBlockSize, p2pconfig.DefaultExcessiveBlockSize)
+	viper.SetDefault(EnvP2PTrickleInterval, p2pconfig.DefaultTrickleInterval)
+	viper.SetDefault(EnvP2PBlocksForForkConfirmation, p2pconfig.DefaultBlocksToConfirmFork)
+}
+
+// ParseP2PConfig init p2p viper config based from specific config file.
+func ParseP2PConfig(cliCfg *p2pconfig.Config) *p2pconfig.Config {
+	viper.SetConfigFile(cliCfg.ConfigFile)
+	if err := viper.ReadInConfig(); err != nil {
+		err = fmt.Errorf("config can't read %s", err.Error())
+		log.Fatal(err)
```

Comment:
I hate this method as it is missleading. IMHO we should have line for logging the error, and second line to panic

Reply:
Could you please clarify? I guess I don't fully understand what you want to say.
Panic will contain this error

Reply:
> Fatal is equivalent to Print() followed by a call to os.Exit(1).

From my perspective this is messing up with purpose of function. Maybe it's cool that it shortens logging and exiting in single line.
But actually if you read it, you can quickly jump over this line, with the thinking that we're only logging here, but it's not true. We're making some big thing, we're shutting down whole program. But it reads like just log a message with Fatal level.

Reply:
fatal replaced with Print() and os.Exit(1)